app/src/main/java/org/p2p/wallet/striga/signup/presetpicker/SelectItemProvider.kt

```diff
@@ -0,0 +1,14 @@
+package org.p2p.wallet.striga.signup.presetpicker
+
+import androidx.recyclerview.widget.RecyclerView
+
+interface SelectItemProvider {
+    fun provideItemsName(): SelectItemsStrings
+
+    fun provideItemDecorations(): List<RecyclerView.ItemDecoration> = emptyList()
+
+    suspend fun provideItems(): List<SelectableItem>
+
+    fun enableSearch(): Boolean
+    fun searchEngine(): SelectItemSearchEngine = DefaultSelectItemSearcher()
```

Comment:
you can create your own Search Engine for universal list and override this method. or you can disable search just by putting false in  enableSearch